### PR TITLE
Specify PING and PONG commands

### PR DESCRIPTION
**Pull request by progval**
Closes GH-59

### CODE REVIEW

**Review by YerinAlexey — APPROVED** [review]
Perfect!

**Comment by YerinAlexey** [comment]
I also noticed that some sections reference `#ping` or `#pong` directly, would you mind fixing that?

**Comment by ProgVal** [comment]
Done.

**Comment by YerinAlexey** [comment]
> Done.

Great, thanks!

**Comment by ProgVal** [comment]
Actually I decided to split it into two sections, as only `PING` triggers these 1 or 3 numerics; and it's more consistent with the other sections.

File: index.md [hunk 1]
```diff
@@ -724,6 +724,46 @@ Command Examples:
                                   ; User gets registered with username
                                   "danp" and real name "Ronnie Reagan"
 
+
+### PING message
+
+         Command: PING
+      Parameters: <token>
```
Comment (emersion) [inline index.md:731]: Maybe this should read `[<token>]` to make it clear it's optional?

Comment (emersion) [reply index.md:731]: Eh, my bad, this is mandatory.

File: index.md [hunk 2]
```diff
@@ -724,6 +724,46 @@ Command Examples:
...
+The `PING` command is sent by either clients or servers to check the other side of the connection is still connected and/or to check for connection latency, at the application layer.
+
+The `<token>` may be any non-empty string.
+
+When receiving a `PING` message, clients or servers must reply to it with a [`PONG`](#pong-message) message with the same `<token>` value. This allows either to match `PONG` with the `PING` they reply to, for example to compute latency.
+
+Clients should not send `PING` during connection registration, though servers may accept it.
+Servers may send `PING` during connection registration and clients must reply to them.
+
+Older versions of the protocol gave specific semantics to the `<token>` and allowed an extra parameter; but these features are not consistently implemented and should not be relied on. Instead, the `<token>`should be treated as an opaque value by the receiver.
```
Comment (delthas) [inline index.md:742]:
```suggestion
Older versions of the protocol gave specific semantics to the `<token>` and allowed an extra parameter; but these features are not consistently implemented and should not be relied on. Instead, the `<token>` should be treated as an opaque value by the receiver.
```

File: index.md [hunk 3]
```diff
@@ -724,6 +724,49 @@ Command Examples:
...
+
+Clients should not send `PING` during connection registration, though servers may accept it.
+Servers may send `PING` during connection registration and clients must reply to them.
+
+Older versions of the protocol gave specific semantics to the `<token>` and allowed an extra parameter; but these features are not consistently implemented and should not be relied on. Instead, the `<token>` should be treated as an opaque value by the receiver.
+
+Numeric Replies:
+
+* [`ERR_NEEDMOREPARAMS`](#errneedmoreparams-461) `(461)`
+* [`ERR_NOORIGIN`](#errnooring-409) `(409)`
```
Comment (emersion) [inline index.md:747]: Typo in anchor: "errnoorigin"

Comment (progval) [reply index.md]: gee, thanks

File: _includes/modern-appendix.md [hunk 4]
```diff
@@ -656,6 +656,13 @@ This is generally sent in response to channel modes, such as a channel being [mo
 
 Indicates that the `JOIN` command failed because the client has joined their maximum number of channels. The text used in the last param of this message may vary.
 
+### `ERR_NOORIGIN (409)`
+
+      ":No origin specified"
```
Comment (emersion) [inline _includes/modern-appendix.md:661]: I think this is missing `<client>`. It's not in the original RFC (where it's implicit) but it's always mentioned in modern-irc.

**Comment by emersion** [comment]
LGTM!

**Pull request merged**